### DISCUSSION TITLE
Why are WhatsApp contact names not shown?

### DISCUSSION THREAD

**Discussion by caltuntas** [d99kris/nchat#114] [Q&A]
This utility is exactly what I was looking for a long time as a terminal user, thanks a lot. I have built and installed on MacOS, everything looks well except contact names. My plan is using it for whatsup. 

![image](https://github.com/d99kris/nchat/assets/35517929/41d856f8-74c4-4b63-9751-2c08e5d8dedc)

as you can see, only mobile phone numbers are displayed as contacts but as far as I can see from the documentation, contact names should be displayed instead of numbers. I double checked on my mobile and whatsapp web client, they are displayed correctly with names. 

Am I missing something? Thanks again.

**Comment by caltuntas** [comment]
I think it was an issue of first time opening. After closing and re-opening it disappeared. Now I can see the names

**Reply by d99kris** [reply]
Good to hear you found a workaround! 

When I find some time I'll try take a look and see if I can reproduce the problem (no contact names after first sign in) and try prepare a fix for it. Will update here.

**Reply by d99kris** [reply]
A fix for whatsapp sync of contact names at first login has been committed in 70e6c02.
